lib/amqp/client/connection.rb

```diff
@@ -190,6 +192,9 @@ def parse_frame(type, channel_id, frame_size, buf)
           when 80 # ack
             delivery_tag, multiple = buf.unpack1("@11 Q> C")
             @channels[channel_id].confirm [:ack, delivery_tag, multiple]
+          when 90 # reject
+            delivery_tag, requeue = buf.unpack1("@11 Q> C")
```

Comment:
Should be unpack here, not unpack1. See the same error on the rows below and over. Unpack1 only when a single element is returned

Reply:
Oh btw, reject will never be sent from the server, only nack will

Reply:
So remove reject here?

Reply:
a5d5941c18fd80fb80ec3b9f579947f66329da9b